### ISSUE TITLE
ghc 9.2.x compatibility

### ISSUE THREAD

**Issue by yaxu** [haskell-hint/hint#138]
hint isn't building on 9.2.1 due to the `ghc>=8.4 && <9.2` constraint:
  https://github.com/tidalcycles/Tidal/runs/4187728052?check_suite_focus=true

Is it enough to simply change the bounds in hint.cabal? I can try it locally if that's helpful.

**Comment by gelisam** [comment]
Yes, currently that's the only change needed; but let's also keep a ticket open to remind us to add a `stack-9.2.yaml` file and a corresponding CI entry once stackage supports that version.

**Comment by brandon-leapyear** [comment]
I don't think that works. I have a branch for adding 9.2 support
https://github.com/haskell-hint/hint/compare/main...brandon-leapyear:ghc-9.2?expand=1

9.2 changes the location of a lot of definitions
